### PR TITLE
Adjust `ESQLRuleData` to Inherit `QueryRuleData` Dataclass

### PR DESCRIPTION
**Pull request by terrancedejesus**
# Related
* https://github.com/elastic/detection-rules/pull/3281
* https://github.com/elastic/detection-rules/pull/3292
* https://github.com/elastic/detection-rules/pull/3294
 
## Summary
After internal discussion and review, it has been determined that moving forward we will remain consistent with our dataclass approach to any rule type with a query, inheriting from `QueryRuleData`. Previously `ESQLRuleData` had been introduced to support core ES|QL rule type, however inherited from `BaseRuleData` since `index` is not valid.

### CODE REVIEW

File: rules/linux/credential_access_credential_dumping_esql.toml
Comment (terrancedejesus) [inline rules/linux/credential_access_credential_dumping_esql.toml:22]: ⚠️ DO NOT MERGE ⚠️

**Review by Mikaayenson — APPROVED** [review]
Looks good! just dont forget to remove the test file. Also dont forget to tweak the unit test.

**Comment by terrancedejesus** [comment]
> Looks good! just dont forget to remove the test file. Also dont forget to tweak the unit test.

Thanks Mika! I chatted with @brokensound77 and he is going to push a commit to fix the testing as he has already solved these that are causing the unit tests errors. Once in and approved, we can merge.

**Comment by brokensound77** [comment]
passing [tests](https://github.com/elastic/detection-rules/actions/runs/7039718211/job/19159266172?pr=3297) with example rule

**Review by brokensound77 — APPROVED** [review]
Updates to unit tests pushed and test rule removed - LGTM 👍

**Comment by terrancedejesus** [comment]
Thanks!

**Pull request merged**
